### PR TITLE
[MFTF] add StorefrontGiftMessageForOrderOnCheckoutCartPageTest 

### LINKED ISSUE #32821: [Issue] [MFTF] add StorefrontGiftMessageForOrderOnCheckoutCartPageTest
This issue is automatically created based on existing pull request: magento/magento2#32792: [MFTF] add StorefrontGiftMessageForOrderOnCheckoutCartPageTest 

---------

### Description (*)

This PR adds `StorefrontGiftMessageForOrderOnCheckoutCartPageTest` 
1. allow gift message for order
2. add product to cart 
3. go to cart page
4. input gift message for order 
5. check gift message  on checkout cart page

### PR DESCRIPTION
**Pull request by Usik2203**
### Description (*)

This PR adds `StorefrontGiftMessageForOrderOnCheckoutCartPageTest` 
1. allow gift message for order
2. add product to cart 
3. go to cart page
4. input gift message for order 
5. check gift message  on checkout cart page 

### Resolved issues:
1. [x] resolves magento/magento2#32821: [MFTF] add StorefrontGiftMessageForOrderOnCheckoutCartPageTest

### CODE REVIEW

**Comment by Usik2203** [comment]
@magento run Functional Tests CE , Functional Tests EE, Functional Tests B2B

**Comment by gabrieldagama** [comment]
@magento create issue

File: app/code/Magento/GiftMessage/Test/Mftf/Test/StorefrontGiftMessageForOrderOnCheckoutCartPageTest.xml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+            <deleteData createDataKey="simpleProduct" stepKey="deleteProduct"/>
+        </after>
+
+        <actionGroup ref="AddSimpleProductToCartActionGroup" stepKey="addProductToCart">
+            <argument name="product" value="$$simpleProduct$$"/>
+        </actionGroup>
+
+        <actionGroup ref="StorefrontCartPageOpenActionGroup" stepKey="goToCartPage" />
+        <actionGroup ref="AssertStorefrontGiftMessageOnCheckoutCartPageActionGroup" stepKey="seeGiftMessageFields"/>
+        <actionGroup ref="StorefrontFieldGiftMessageCartFormActionGroup" stepKey="fillGiftMessageFields"/>
```
Comment (gabrieldagama) [inline app/code/Magento/GiftMessage/Test/Mftf/Test/StorefrontGiftMessageForOrderOnCheckoutCartPageTest.xml:33]: Can we add few more steps to place the order and check the messages are shown on the admin?

**Review by gabrieldagama — CHANGES_REQUESTED** [review]

**Comment by Usik2203** [comment]
@magento run Functional Tests CE , Functional Tests EE, Functional Tests B2B

**Comment by Usik2203** [comment]
@magento run Functional Tests CE , Functional Tests EE, Functional Tests B2B

**Comment by Usik2203** [comment]
@magento give me 2.4-develop instanc

**Comment by Usik2203** [comment]
@magento give me 2.4-develop instance

**Comment by Usik2203** [comment]
@magento run Functional Tests CE , Functional Tests EE, Functional Tests B2B

**Comment by Usik2203** [comment]
@magento run all tests

**Comment by Usik2203** [comment]
@magento run all tests

**Comment by engcom-Charlie** [comment]
Hi @Usik2203,

Thanks for your contribution!

I am working on the requested changes mentioned by @gabrieldagama

**Comment by engcom-Charlie** [comment]
@gabrieldagama, It seems that the test case has been modified as per your review comments by adding few Action groups and Section in test case. 

The modified test case is running fine so moving it for review.

**Review by sidolov — APPROVED** [review]

**Pull request merged**